src/app/services/editors/gameobject/gameobject-template.service.spec.ts

```diff
@@ -17,7 +19,14 @@ describe('GameobjectTemplateService', () => {
   }));
 
   it('should be created', () => {
-    const service: GameobjectTemplateService = TestBed.get(GameobjectTemplateService);
-    expect(service).toBeTruthy();
+    const gameobjectService: GameobjectTemplateService = TestBed.get(GameobjectTemplateService);
+    expect(gameobjectService).toBeTruthy();
+  });
+
+  it('changing type and the related fields data*', () => {
+    const gameobjectService: GameobjectTemplateService = TestBed.get(GameobjectTemplateService);
+    expect(gameobjectService.getFieldDefinition(0, 0)).toBe(GO_DATA_FIELDS[0][0]);
+    expect(gameobjectService.getFieldDefinition(0, 25)).toEqual({ name: `Data25`, tooltip: null });
```

Comment:
ideally those should be 2 separate test cases, something like:

- getFieldDefinition() should correctly return the definition according to the type
- getFieldDefinition() should correclty return a default value when the definition is missing

but the logic is good